src/Authoring/WinRT.SourceGenerator/WinRTTypeWriter.cs

```diff
@@ -1,2180 +1,2682 @@
+                    SignatureCallingConvention.Default,
+                    0,
+                    !isStatic)
+                .Parameters(
+                    parameters.Length,
+                    returnType => EncodeReturnType(returnSymbol, returnType),
+                    parametersEncoder => EncodeParameters(parameters, parametersEncoder)
+                );
+
+            List<ParameterHandle> parameterHandles = new List<ParameterHandle>();
+            for (int idx = 0; idx < parameters.Length; idx++)
+            {
+                parameterHandles.Add(metadataBuilder.AddParameter(
+                    parameters[idx].Attributes,
+                    metadataBuilder.GetOrAddString(parameters[idx].Name),
+                    idx + 1));
+            }
+
+            var methodAttributes =
+                (isPublic ? MethodAttributes.Public : MethodAttributes.Private) |
+                MethodAttributes.HideBySig;
+
+            var methodImplAttributes = MethodImplAttributes.Managed;
+
+            if (isInterfaceParent)
+            {
+                methodAttributes |= MethodAttributes.Abstract;
+            }
+            else
+            {
+                // TODO check if from overridable interface
+                if (!isOverridable)
+                {
+                    methodAttributes |= MethodAttributes.Final;
+                }
+                methodImplAttributes |= MethodImplAttributes.Runtime;
+            }
+
+            if (isSpecialMethod && name == ".ctor")
+            {
+                methodAttributes |= MethodAttributes.RTSpecialName;
+            }
+            else if(isStatic)
+            {
+                methodAttributes |= MethodAttributes.Static;
+            }
+            else
+            {
+                methodAttributes |=
+                    MethodAttributes.Virtual |
+                    MethodAttributes.NewSlot;
+            }
+
+            if (isSpecialMethod)
+            {
+                methodAttributes |= MethodAttributes.SpecialName;
+            }
+
+            var methodDefinitionHandle = metadataBuilder.AddMethodDefinition(
+                methodAttributes,
+                methodImplAttributes,
+                metadataBuilder.GetOrAddString(name),
+                metadataBuilder.GetOrAddBlob(methodSignature),
+                -1,
+                parameterHandles.Count == 0 ?
+                    MetadataTokens.ParameterHandle(metadataBuilder.GetRowCount(TableIndex.Param) + 1) :
+                    parameterHandles[0]);
+            return methodDefinitionHandle;
+        }
+
+        public override void VisitMethodDeclaration(MethodDeclarationSyntax node)
+        {
+            Logger.Log("method: " + node.Identifier.ValueText);
+            var methodSymbol = Model.GetDeclaredSymbol(node);
+            if (!IsPublicNode(node) || currentTypeDeclaration.CustomMappedSymbols.Contains(methodSymbol))
+            {
+                Logger.Log("method skipped");

```

Comment:
do we want to say the name? might help debugging

Reply:
That is logged 3 lines ahead (Logger.Log("method: " + node.Identifier.ValueText);)